src/libs/ajax/ajax-common.ts

```diff
@@ -35,6 +35,29 @@ export const withRetryOnError = _.curry((shouldNotRetryFn, wrappedFetch) => asyn
   return wrappedFetch(...args);
 });
 
+const TIMEOUT_DURATION = 10000;
+
+export function makeRequestRetry(request: Function, retriesLeft: number) {
```

Comment:
can we put the return value on this function? Presumably its returning a promise, so lets enforce the caller to await it